genie-web/src/test/groovy/com/netflix/genie/web/scripts/ScriptManagerSpec.groovy

```diff
@@ -0,0 +1,469 @@
+        compiledScriptReference == loadScriptTask.compiledScriptReference
+
+        when: "Attempt to load"
+        loadScriptTask.run()
+
+        then:
+        0 * resourceLoader.getResource(scriptUri.toString())
+        1 * meterRegistry.timer(
+            ScriptManager.SCRIPT_LOAD_TIMER_NAME,
+            { Set<Tag> tags -> tags.containsAll(loadingErrorTags) }
+        ) >> timer
+        1 * timer.record({ it > 0 }, TimeUnit.NANOSECONDS)
+    }
+
+    def "evaluateScript"() {
+        URI scriptUri = new URI("s3://genie/scripts/myscript.js")
+        CompiledScript compiledScript = Mock(CompiledScript)
+        Throwable evaluationException = new TimeoutException("...")
+        Object evaluationResult = new Object()
+
+        Timer timer = Mock(Timer)
+        Set<Tag> evalErrorTags = MetricsUtils.newFailureTagsSetForException(evaluationException)
+        Set<Tag> evalSuccessTags = MetricsUtils.newSuccessTagsSet()
+
+        Tag uriTag = Tag.of(MetricsConstants.TagKeys.SCRIPT_URI, scriptUri.toString())
+        evalErrorTags.add(uriTag)
+        evalSuccessTags.add(uriTag)
+
+        Bindings bindings = Mock(Bindings)
+        long timeout = 5_000L
+        Future<Object> evalTaskFuture = Mock(Future)
+
+        when: "Evaluate unknown script"
+        this.scriptManager.evaluateScript(scriptUri, bindings, timeout)
+
+        then:
+        thrown(ScriptNotConfiguredException)
+
+        when: "Evaluate script not yet compiled successfully"
+        this.scriptManager.scriptsMap.put(scriptUri, new AtomicReference<>(null))
+        this.scriptManager.evaluateScript(scriptUri, bindings, timeout)
+
+        then:
+        thrown(ScriptNotConfiguredException)
+
+        when:
+        this.scriptManager.scriptsMap.put(scriptUri, new AtomicReference<>(compiledScript))
+        this.scriptManager.evaluateScript(scriptUri, bindings, timeout)
+
+        then:
+        1 * executorService.submit(_) >> evalTaskFuture
+        1 * evalTaskFuture.get(timeout, TimeUnit.MILLISECONDS) >> { throw evaluationException }
+        1 * evalTaskFuture.cancel(true)
+        1 * meterRegistry.timer(
+            ScriptManager.SCRIPT_EVALUATE_TIMER_NAME,
+            {
+                Set<Tag> tags -> tags.containsAll(evalErrorTags)
+            }
+        ) >> timer
+        1 * timer.record({ it > 0 }, TimeUnit.NANOSECONDS)
+        thrown(ScriptExecutionException)
+
+        when:
+        def result = this.scriptManager.evaluateScript(scriptUri, bindings, timeout)
+
+        then:
+        1 * executorService.submit(_) >> evalTaskFuture
+        1 * evalTaskFuture.get(timeout, TimeUnit.MILLISECONDS) >> evaluationResult
+        1 * meterRegistry.timer(
+            ScriptManager.SCRIPT_EVALUATE_TIMER_NAME,
+            {
+                Set<Tag> tags -> tags.containsAll(evalSuccessTags)
+            }
+        ) >> timer
+        1 * timer.record({ it > 0 }, TimeUnit.NANOSECONDS)
+        result == evaluationResult
+    }
+//
```

Comment:
A lot of commented out code here. If you have a test you want to skip there's a way to do that with spock

Reply:
Deleted, didn't intend to commit commented out code.